### ISSUE TITLE
About label['roidb']['segmes']

### ISSUE THREAD

**Issue by JoyHuYY1412** [uber-research/UPSNet#53]
HI, for coco dataset, when I print label['roidb']['segmes'] in resnet_upsnet.py, I found sometimes the last segmentation is not polygon format, which is of list format, but they look like RLE format, for example, a dict {"counts': ,'size':}.
 What's more, the counts part is not binary string but polygon list, which is really confusing.

I have no idea why this could happen, could you please help me?

**Comment by JoyHuYY1412** [comment]
I give some example:

> 
> {'counts': [20939, 13, 289, 18,
>  12, 1, 271, 35, 7, 12, 248, 56, 245, 59, 242, 62, 218, 1, 19, 65, 217, 2, 17, 67, 216, 4, 14, 69, 216, 4, 13, 70, 216, 5, 12, 70, 216, 6, 11, 70, 216, 6, 11, 70, 216, 7, 11, 68, 217, 8, 10, 67, 218, 8, 1
> 0, 65, 220, 9, 9, 56, 229, 10, 8, 57, 228, 10, 9, 57, 227, 11, 8, 61, 223, 12, 8, 63, 220, 13, 7, 64, 220, 13, 7, 64, 220, 13, 6, 67, 219, 12, 6, 67, 220, 11, 6, 67, 221, 10, 5, 68, 225, 6, 5, 25, 9, 34,
> 223, 8, 4, 22, 13, 33, 223, 8, 5, 16, 19, 32, 223, 8, 5, 15, 20, 32, 223, 8, 5, 2, 1, 11, 22, 31, 223, 8, 5, 2, 5, 5, 25, 30, 223, 9, 4, 2, 37, 28, 223, 9, 5, 1, 39, 25, 224, 10, 4, 1, 47, 16, 225, 15, 53
> , 8, 227, 15, 288, 15, 288, 15, 288, 15, 288, 14, 289, 13, 290, 12, 290, 13, 290, 13, 290, 13, 290, 13, 290, 12, 290, 13, 290, 13, 290, 13, 290, 13, 290, 13, 289, 13, 290, 13, 289, 14, 289, 14, 288, 15, 2
> 88, 15, 287, 15, 288, 15, 287, 16, 287, 15, 288, 15, 287, 12, 291, 11, 292, 9, 294, 8, 295, 8, 295, 8, 295, 7, 297, 6, 298, 5, 300, 2, 5750, 2, 300, 4, 299, 4, 299, 5, 298, 5, 297, 7, 296, 7, 296, 8, 295,
>  8, 295, 9, 293, 10, 293, 11, 292, 11, 292, 12, 291, 12, 291, 13, 290, 13, 4, 1, 285, 14, 2, 2, 285, 14, 1, 3, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286
> , 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 17, 286, 18, 285, 18, 286, 17, 287, 16, 289, 14, 290, 13, 290, 13, 291, 11, 293, 9, 296, 5, 48167, 6, 2
> 96, 8, 294, 10, 292, 12, 291, 12, 34, 7, 250, 12, 1, 11, 21, 9, 249, 29, 15, 11, 248, 33, 10, 13, 247, 34, 9, 13, 247, 35, 8, 13, 247, 37, 5, 14, 247, 38, 2, 16, 247, 56, 247, 56, 247, 56, 247, 56, 247, 5
> 6, 247, 56, 247, 56, 247, 56, 247, 56, 247, 56, 247, 56, 247, 55, 248, 55, 248, 54, 249, 54, 249, 54, 249, 14, 5, 35, 250, 12, 11, 30, 251, 10, 19, 23, 253, 7, 26, 17, 293, 10, 294, 8, 296, 6, 298, 5, 298
> , 4, 299, 4, 299, 4, 299, 4, 299, 3, 300, 2, 300, 1, 26312], 'size': [303, 500]}]




> 
> [{'size': [375, 500], 'counts': b'm_18\\;5L3N2M2N2O1O100000O1OL0WE[Oh:d06O1100000NKAVE>i:FUE:j::TGXOh6i0VI[Oh6f0RIAl6`0QICn62_GGa1:n6NaGJ_1:o6:QIGn69RIHm65VIKi65XIKh65XIKh64YILg65YIJg66ZIIf6b0PI]OP71eGJ\\14o62hGG\\14l65fGI_10l67dGJc1Kj6m0ZInNg6;hGMX:MnE3R:MnE4Q:KQF4o9JSF6m9HUF8l9D\\F7c:O00M3O1N2BCaE>^:=1000000O100O1O2N2N1N3N2L4N2M^_1Kg`N4N100K4N200O1FB[E>d:FXE;h::0O000100QF_Oa8JdFh0i0B_8f0`G]O]8d0bG^O\\8c0cG@[8`0eGBY8>gGDW8<iGEV8;jGFU8:kGFU8:kGGT89lGGU88kGHU87YGTO5e0c86VGYO5a0f85TG]O4>h84TG@3<j82TGC1;k81TGGN9n8@dGW1]8hNcGX1]8hNbGX1_8hN`GY1`8gNZG^1g8cNVG^1l8bNRG^1n8cNQGn00lNo8;mFi0b9f00O01O10O1O11_FQNX9Y2N1O\\NhFj0W9VOjFi0V9WOkFh0U9XOkFh0V9VOkFj0V9TOlFk0U9SOlFm0V9QOjFV1n8lNQGV1l8kNTGV1j8kNVGV1h8kNXGU1g8lNZGS1e8nN\\GQ1d8oN]GP1c8PO]GP1c8POWGV1i8jNUGW1l8i01O2ROUG`NZ9^1fFdNZ9Z1gFfNZ9X1gFhNY9W1hFiN1Ij8W1[GPOIMj8S1]GROFMl88RG3=JBMo85SGE21;f0a8B`GF0h0X8J_H6`7JbH5^7JcH6]7JcH6]7IeH5]7IeH5\\7JfH4[7MgHOZ73eHK\\76eHG\\7:oG]OB7`8<THZOf8e0kFD[9<[FZOIa0i96]F[OGc0j92]F7`9I_F9`9F`F<6nNk8e0PG`02mNl8b0SGe0LkNP9>VGR1i8mNXGS1h8lNYGT1g8lNYGT1g8lNYGT1g8lNYG
> T1g8lNYG<EIQ9LZG:IFn80YG7MGk82XGJ;2^84WGJo96RFHo98RFFo9:RFDo9<f01O1O000000000WF@[8`0cGD[8<dGF[8:dGH[88\\GCUO5_98ZG3f8MYG4h8KWG6i8JWG:e8F[G>a8B_G`0_8@aGa0]8@cGa0\\8_OdGa0\\8_OeG`0[8@fG?Z8AgG>Y8BjG;V8ElG9T8
> GlG8U8HkG7V8HkGITOLR9;jGId86^GHd87]GGd88^GFc8:]1O11O1O10O00O1000000O1000000cF^Oe7b0RH^OjN1S9a0UH^OfN2T9`0iG^OXO1K1T9`0hGDPO12KV9`0hG0oNAY9?hG5X8JiG6W8JiG6W8JiG5X8KgG5Z8KdG6]8JbG6_8IaG9^8FcG:]8EdG;\\8CfG=Z
> 8BfG?Z8@gG`0Y8@gG`0Y8AfG?Z8AeG`0[8@cGa0^8_OaGa0`8_O_Ga0b8_O^G`0c8@]Ga0b8^O_Gc0`8]O`Gd0_8[ObGe0^8[ObGe0^8[OcGd0]8[OeGd0[8\\OfGc0Z8]OhGa0X8_OiG?X8@kG=V8BlG<U8CmG;Y`2DViMg0L3N2N100000000000O2N2N2MYS1[O\\mN4M
> 2N2O00000O100000O0000010O1M3O10O011N4L001O10002N3RE_O_:o0N1O1N100000000000000000O100O2N2N2O7IO1O1O1O10000000000000O1O001N20000000001O1N2N3MO012N2N2N1O10000000000000000001O1N2N7J0O01O10000O101O0O1O1000K5N2
> O1OGOnD1k:8TEGj:<VECi:?WE@h:a071O00001O004M1O0006KN1O1O0001O1O1O4L8G2N3MUci0'}]]

**Comment by LaoYang1994** [comment]
@JoyHuYY1412 Hi, I met this problem when using this code. That's because the "crowd instances" are included. You can refer to #33 .

Please let me know if you have any problem.

**Comment by JoyHuYY1412** [comment]
> @JoyHuYY1412 Hi, I met this problem when using this code. That's because the "crowd instances" are included. You can refer to #33 .
> 
> Please let me know if you have any problem.

Yes, after I edited my files as https://github.com/uber-research/UPSNet/commit/ba524d5535e47ea42216dc299fa3effad4e3e661 did, but unfortunately the performance of panoptic head decreased, for example, the panoptic loss becomes larger and accuracy becomes smaller. Have you encountered this problem?

**Comment by LaoYang1994** [comment]
@JoyHuYY1412 I am sorry that I can't remember exactly. But I remember that the performance seems to be worse. 
The pan_loss's becoming larger is normal. Because when you drop the crowd instance, the gt of this region will be wrong, which may affect the loss.

Please let me know if you have any problem.

**Comment by JoyHuYY1412** [comment]
> @JoyHuYY1412 I am sorry that I can't remember exactly. But I remember that the performance seems to be worse.
> The pan_loss's becoming larger is normal. Because when you drop the crowd instance, the gt of this region will be wrong, which may affect the loss.
> 
> Please let me know if you have any problem.

Yes, I believe when I drop the crowd instance, the gt of the crowd part will also be "unknown", but it also makes sense when we calculate loss, right? So I was confused when the accuracy drops so much.

**Comment by LaoYang1994** [comment]
@JoyHuYY1412 
Oh, you are right. I was wrong.  It's hard to say. It seems that we need the author's help. @YuwenXiong

**Comment by JoyHuYY1412** [comment]
> @JoyHuYY1412
> Oh, you are right. I was wrong. It's hard to say. It seems that we need the author's help. @YuwenXiong

Thank you for your reply, but I still don't quiet understand why there will exist these kind of format in lable['roidb']['segms']. Especially the first one, it is not RLE or Polygen, moreover, the code here https://github.com/uber-research/UPSNet/blob/3218581a623b02a73c3334b672fc1ce0c25fdae9/upsnet/dataset/json_dataset.py#L173-L179 also filtered some data, right?
